### PR TITLE
test: clarify the support for IPV6

### LINKED ISSUE #9378: docs: Stream Route's `remote_addr` and `server_addr` should support IPv6 as well
### Current State

In [Admin API reference, Stream Route](https://apisix.apache.org/docs/apisix/admin-api/#stream-route), `remote_addr` and `server_addr` should support both IPv4 and IPv6. Currently only the IPv4 usage is documented and **tested**  

Tests for stream route can be found at `t/admin/stream-routes.t` and `t/admin/stream-routes-disable.t`

### Desired State

`remote_addr` and `server_addr` seem to support IPv6 as well and should be added to the above doc.

For a more rigorous update, **test cases** should be added as well. (so not sure if this issue should be labeled as `doc`)

### Reasoning

In `apisix/stream/router/ip_port.lua`:
```
if item.value.remote_addr then
    item.value.remote_addr_matcher = core_ip.create_ip_matcher({item.value.remote_addr})
end
if item.value.server_addr then
    item.value.server_addr_matcher = core_ip.create_ip_matcher({item.value.server_addr})
end
```
where `core_ip` is `local core_ip  = require("apisix.core.ip")`

In `apisix/core/ip`:
```
local ipmatcher = require("resty.ipmatcher")
...

function _M.create_ip_matcher(ip_list)
    local ip, err = ipmatcher.new(ip_list)
    if not ip then
        log.error("failed to create ip matcher: ", err,
                  " ip list: ", json.delay_encode(ip_list))
        return nil
    end

    return ip
end
```
where `ipmatcher` can be sourced back to [lua-resty-ipmatcher](https://github.com/api7/lua-resty-ipmatcher)

### PR DESCRIPTION
**Pull request by shreemaan-abhishek**
### Description

Fixes #9378 

### Checklist

- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: docs/en/latest/admin-api.md [hunk 1]
```diff
@@ -1382,8 +1382,8 @@ Stream Route resource request address:  /apisix/admin/stream_routes/{id}
 | ----------- | -------- | -------- | ------------------------------------------------------------------- | ----------------------------- |
 | upstream    | False    | Upstream | Configuration of the [Upstream](./terminology/upstream.md). |                               |
 | upstream_id | False    | Upstream | Id of the [Upstream](terminology/upstream.md) service.      |                               |
-| remote_addr | False    | IP/CIDR  | Filters Upstream forwards by matching with client IP.               | "127.0.0.1/32" or "127.0.0.1" |
-| server_addr | False    | IP/CIDR  | Filters Upstream forwards by matching with APISIX Server IP.        | "127.0.0.1/32" or "127.0.0.1" |
+| remote_addr | False    | IP/CIDR  | Filters Upstream forwards by matching with client IP.               | "127.0.0.1/32" or "127.0.0.1" or "::ffff:7f00:1" |
```
Comment (kayx23) [inline docs/en/latest/admin-api.md:1385]:
```suggestion
| remote_addr | False    | IPv4, IPv4 CIDR, IPv6  | Filters Upstream forwards by matching with client IP.               | "127.0.0.1" or "127.0.0.1/32" or "::ffff:7f00:1" |
```

File: t/admin/stream-routes.t [hunk 2]
```diff
@@ -607,3 +607,47 @@ GET /t
...
+
+=== TEST 17: set route with remote_addr and server_addr in IPV6
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/stream_routes/1',
+                ngx.HTTP_PUT,
+                [[{
+                    "remote_addr": "::ffff:7f00:1",
```
[the remark this answers is not in the document]
Comment (shreemaan-abhishek) [reply t/admin/stream-routes.t:621]: Oh, I did not know that a simpler notation existed.

[the remark this answers is not in the document]
Comment (kayx23) [reply t/admin/stream-routes.t]: look at my profile bio? location? lol.

If I recall correctly, `::ffff:7f00:1` is IPv4-mapped. `::1` is the most native and standard IPv6 loopback address.

**Review by kayx23 — APPROVED** [review]
looks good to me

**Comment by Sn0rt** [comment]
LGTM

**Comment by kayx23** [comment]
cc @leslie-tsang

**Review by navendu-pottekkat — APPROVED** [review]

**Review by monkeyDluffy6017 — APPROVED** [review]

**Pull request merged**